pkg/table/tables/partition.go

```diff
@@ -1312,30 +1317,46 @@ func (t *partitionedTable) locatePartitionCommon(ctx expression.EvalContext, tp
 		} else {
 			idx, err = t.locateRangePartition(ctx, partitionExpr, r)
 		}
+		if err != nil {
+			return -1, err
+		}
+		pi := t.Meta().Partition
+		if pi.CanHaveOverlappingDroppingPartition() {
+			pid := pi.Definitions[idx].ID
+			for _, droppingDef := range pi.DroppingDefinitions {
```

Comment:
It seems better to create a function for this operation like `pi.IsDropping(idx)` to help simplify this code.

Reply:
Done, thanks for the suggestion!

Reply:
oh, this place is not using it yet, fixing...